src/server/json_family.cc

```diff
@@ -51,6 +52,64 @@ using CI = CommandId;
 
 namespace {
 
+// This class should be used before any json is allocated within a flow and must be destructed
+// last. If it's constructed first locally it will be destructed last as the order of
+// destructors in C++ is the opposite of constructors. If this condition is ever violated
+// the memory tracking will be off, because we must first destruct the temporaries
+// created in json flows and then calculate how much memory is left and assign that to the
+// json object within the CompactObject to get an accurate result.
+// It uses RAII to set the memory usage of json objects.
+class JsonMemTracker {
+ public:
+  explicit JsonMemTracker(const OpArgs& args, std::string_view key, bool op_set = false)
+      : op_args_(args), key_(key), op_set_{op_set} {
+    start_size_ = static_cast<MiMemoryResource*>(CompactObj::memory_resource())->used();
+    auto it_res = op_args_.GetDbSlice().FindMutable(op_args_.db_cntx, key_, OBJ_JSON);
```

Comment:
we now call Find 3 times more times?

Reply:
2 more times in the worst case (when the json does not exist). Otherwise it's one.

Reply:
is it possible to use a similar pattern with already existing iterator? i.e. avoid additional finds?

Reply:
Can we modify `JsonMemTracker` to use the compact object directly (i.e. not to call `Find()`) and modify `UpdateEntry()` to use it?
And even if I missed some call that it won't work for, we can code around that instead, because it does seem like all (the majority?) of calls use it